core/modulesource.go

```diff
@@ -380,7 +380,30 @@ func (src *ModuleSource) LoadContext(ctx context.Context, dag *dagql.Server, pat
 			return inst, fmt.Errorf("failed to get dir instance: %w", err)
 		}
 
-		return loadedDir, nil
+		if ignore == nil || len(ignore) == 0 {
+			return loadedDir, nil
+		}
+
+		var ignoredDir dagql.Instance[*Directory]
+
+		err = dag.Select(ctx, dag.Root(), &ignoredDir,
+			dagql.Selector{
+				Field: "directory",
+			},
+			dagql.Selector{
+				Field: "withDirectory",
+				Args: []dagql.NamedInput{
+					{Name: "path", Value: dagql.String("/")},
+					{Name: "directory", Value: dagql.NewID[*Directory](loadedDir.ID())},
+					{Name: "exclude", Value: dagql.ArrayInput[dagql.String](dagql.NewStringArray(ignore...))},
```

Comment:
We should remove the `exclude`/`include` vars before the switch statement if we're doing it this way.

Reply:
True

Reply:
Done in [21f7519](https://github.com/dagger/dagger/pull/8430/commits/21f75197f329ce56cf4dfd3d4c8a86a25d71ea0a)